### PR TITLE
Add segment attributes to the `_segments` API.

### LINKED ISSUE #26130: Provide index level indication when index.codec is set at the node level
https://www.elastic.co/guide/en/elasticsearch/reference/5.0/breaking_50_settings_changes.html#_index_level_settings

Starting in 5.0, most index level settings cannot be set at the node level anymore, except for a few like `index.codec`.

The user experience of applying this at the node level can be improved.  For example, say an admin added `index.codec: best_compression` to the node configurations but don't remember when this was added.  What this means is the new indices created after `index.codec` is added should be using best compression.  But there isn't a good way for admins to tell because if they do a GET _settings against an index that is created after the node is started with index.codec, the index settings will not show that the index has index.codec configured.   They can only tell that index.codec is set from the nodes info api (but that's at the node level).    

It will be helpful we can apply index.codec to the index settings for indices that are created while the nodes have index.codec set (and if it shows something different, then it will just imply that the index settings have been updated via a direct index settings call post index creation).  Or provide some indication from the segments api to show if the stored fields are compressed (using best compression)?

### PR DESCRIPTION
**Pull request by jpountz**
This contains information about whether high compression was enabled for instance.

Closes #26130

### CODE REVIEW

**Comment by jpountz** [comment]
@elasticmachine test it please

File: core/src/main/java/org/elasticsearch/index/engine/Segment.java [hunk 1]
```diff
@@ -173,6 +183,11 @@ public void readFrom(StreamInput in) throws IOException {
         } else {
             segmentSort = null;
         }
+        if (in.getVersion().onOrAfter(Version.V_6_1_0) && in.readBoolean()) {
+            attributes = in.readMap(StreamInput::readString, StreamInput::readString);
+        } else {
+            attributes = null;
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/index/engine/Segment.java]: Why not default this to `emptyMap()` so you don't need a null check in `toXContent`?

**Review by rjernst — APPROVED** [review]
LGTM

File: core/src/main/java/org/elasticsearch/index/engine/Segment.java (hunk 1 above)
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/engine/Segment.java]: I wanted to have a difference between empty attributes and unkwown attributes, at least internally

**Comment by jpountz** [comment]
@elasticmachine test it please

**Pull request merged**